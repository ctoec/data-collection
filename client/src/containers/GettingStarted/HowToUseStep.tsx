import React from 'react';
import { TextWithIcon } from '@ctoec/component-library';
import { ReactComponent as Arrow } from '@ctoec/component-library/dist/assets/images/arrowRight.svg';

type HowToUseStepProps = {
  stepHeader: string;
  paragraphText: string;
  buttonText: string;
  linkTo: string;
};

export const HowToUseStep: React.FC<HowToUseStepProps> = ({
  stepHeader,
  paragraphText,
  buttonText,
  linkTo,
}) => {
  return (
    <>
      <p className="how-to-header">{stepHeader}</p>
      <p className="how-to-body">{paragraphText}</p>
      <p className="how-to-button">
        {/* Need to use an a tag to get href for an external link,
            since the how-to guide is hosted on documize */}
        <a className="usa-button usa-button--unstyled" href={linkTo}>
          <TextWithIcon text={buttonText} Icon={Arrow} iconSide="right" />
        </a>
      </p>
    </>
  );
};
