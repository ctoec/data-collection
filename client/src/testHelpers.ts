import React from 'react';
import { RenderResult, render, cleanup, act } from '@testing-library/react';
import { axe } from 'jest-axe';

export const accessibilityTestHelper = (
  component: React.ReactElement,
  before?: (_: RenderResult) => Promise<void>
) => {
  it('passes AXE accessibility checks', async () => {
    await act(async () => {
      const renderResult = render(component);
      if (before) {
        await before(renderResult);
      }
      const results = await axe(renderResult.container);
      expect(results).toHaveNoViolations();
    });
    cleanup();
  });
};

export const snapshotTestHelper = (component: React.ReactElement) => {
  it('matches snapshot', async () => {
    await act(async () => {
      const { asFragment } = render(component);
      expect(asFragment()).toMatchSnapshot();
    });
  });
};
