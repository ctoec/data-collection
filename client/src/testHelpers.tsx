import React from 'react';
import { RenderResult, render, cleanup, act } from '@testing-library/react';
import { axe } from 'jest-axe';
import { BrowserRouter } from 'react-router-dom';

export const renderHelper = async (
  component: React.ReactElement,
  opts: {
    before?: (_: RenderResult) => Promise<void>;
    wrapInRouter?: boolean;
  } = {}
) => {
  const renderResult = render(
    opts.wrapInRouter ? <BrowserRouter>{component}</BrowserRouter> : component
  );

  if (opts.before) {
    await opts.before(renderResult);
  }
  return renderResult;
};

export const accessibilityTestHelper = (
  component: React.ReactElement,
  opts: {
    before?: (_: RenderResult) => Promise<void>;
    wrapInRouter?: boolean;
  } = {}
) => {
  it('passes AXE accessibility checks', async () => {
    await act(async () => {
      const renderResult = await renderHelper(component, opts);
      const results = await axe(renderResult.container);
      expect(results).toHaveNoViolations();
    });

    cleanup();
  });
};

export const snapshotTestHelper = (
  component: React.ReactElement,
  opts: {
    before?: (_: RenderResult) => Promise<void>;
    wrapInRouter?: boolean;
  } = {}
) => {
  it('matches snapshot', async () => {
    await act(async () => {
      const renderResult = await renderHelper(component, opts);
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    cleanup();
  });
};
