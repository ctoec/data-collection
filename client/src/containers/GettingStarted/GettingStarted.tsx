import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import { getH1RefForTitle } from '../../utils/getH1RefForTitle';
import { ReactComponent as Image } from '../../images/PersonWithSpreadsheet.svg';
import UserContext from '../../contexts/UserContext/UserContext';
import { HowToUseStep } from './HowToUseStep';

const GettingStarted: React.FC = () => {
  const { user } = useContext(UserContext);
  const h1Ref = getH1RefForTitle();
  const org = (user?.organizations || [])[0] || {};

  return (
    <div className="GettingStarted grid-container margin-top-4">
      <div className="grid-row grid-gap">
        <div className="tablet:grid-col-8">
          <h1 ref={h1Ref}>Hello {user?.firstName}!</h1>
          <p>
            ECE Reporter will allow you to share your state-funded enrollment
            data with the Office of Early Childhood (OEC). OEC uses this data to
            make informed program and policy decisions affecting young children
            and families.
          </p>
          <h2>Sites you manage at {org.providerName}:</h2>
          <ul>
            {(user?.sites || []).map((site) => (
              <li key={site.siteName} className="line-height-body-4">
                {site.siteName}
              </li>
            ))}
          </ul>
          <h2>How to use ECE Reporter</h2>
          <p>
            {
              "These instructions guide you through the file upload process to add many records at once. If you'd like to add records manually, "
            }
            <Link to="/roster/"> you can add them on your roster</Link>.
          </p>
          <ol className="how-to-list">
            <li key="review-requirements" className="how-to-step">
              <HowToUseStep
                stepHeader="Review the data requirements"
                paragraphText="Learn more about the data elements ECE Reporter collects for each enrollment."
                buttonText="See required data"
                linkTo="/data-requirements/"
              />
            </li>
            <li key="download-template" className="how-to-step">
              {
                <HowToUseStep
                  stepHeader="Download the data template"
                  paragraphText="You'll need to enter data in this file to upload in ECE Reporter."
                  buttonText="Download the template"
                  linkTo="/template/"
                />
              }
            </li>
            <li key="prepare-data" className="how-to-step">
              {
                <HowToUseStep
                  stepHeader="Prepare your data"
                  paragraphText="Get instructions on how to enter data into OEC's template."
                  buttonText="See the how-to guide"
                  linkTo="https://ece-reporter.documize.com/s/burrnubmbdja9sqnbh6g/ece-reporter-help/d/bv34kdhmdesjli7los1g/how-to-prepare-your-data-for-ece-reporter"
                />
              }
            </li>
            <li key="upload" className="how-to-step">
              {
                <HowToUseStep
                  stepHeader="Upload your data"
                  paragraphText="Once your template is filled out, you'll upload the data to ECE Reporter."
                  buttonText="Go to file upload"
                  linkTo="/upload/"
                />
              }
            </li>
          </ol>
        </div>
        <div className="tablet:grid-col-4" role="presentation">
          <Image />
        </div>
      </div>
    </div>
  );
};

export default GettingStarted;
