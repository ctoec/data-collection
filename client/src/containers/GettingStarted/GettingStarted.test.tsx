import React from 'react';
import GettingStarted from './GettingStarted';
import { snapshotTestHelper, accessibilityTestHelper } from '../../testHelpers';

describe('GettingStarted', () => {
  snapshotTestHelper(<GettingStarted />, { wrapInRouter: true });
  accessibilityTestHelper(<GettingStarted />, { wrapInRouter: true });
});
