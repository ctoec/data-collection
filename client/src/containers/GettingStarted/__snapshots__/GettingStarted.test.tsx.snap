// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GettingStarted matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="grid-container margin-top-4"
  >
    <div
      class="grid-row grid-gap"
    >
      <div
        class="grid-col-8"
      >
        <h1>
          Let's get started
        </h1>
        <p>
          This tool will allow you to share your state-funded enrollment data with OEC.
        </p>
        <h2>
          Here's what to include
        </h2>
        <h3>
          Enrollments in spaces funded by
        </h3>
        <ul>
          <li>
            Child Day Care (CDC)
          </li>
          <li>
            Priority or Competitive School Readiness (PSR and CSR)
          </li>
          <li>
            Smart Start (SS)
          </li>
          <li>
            State Head Start (SHS)
          </li>
        </ul>
        <div
          class="usa-alert usa-alert--info grid-row"
        >
          <div
            class="display-flex flex-fill"
          >
            <div
              class="usa-alert__body grid-col flex-fill"
            >
              <h2
                class="usa-alert__heading"
              >
                Learn more about required data
              </h2>
              <p
                class="usa-alert__text"
              >
                <span>
                  See all required data fields, definitions and rationale in 
                  <a
                    class="usa-button usa-button--unstyled"
                    href="/data-requirements"
                  >
                    OEC's enrollment data requirements
                  </a>
                </span>
              </p>
            </div>
            <div
              class="usa-alert__body flex-auto display-flex flex-align-center"
            />
          </div>
        </div>
        <h3>
          Reporting period
        </h3>
        <p>
          Data is due on March 31, 2020 and must include data from July 1, 2020 to at least Jan 1, 2021.
        </p>
        <div>
          <h2>
            Download the latest batch upload template
          </h2>
          <button
            class="usa-button usa-button--unstyled text-bold margin-bottom-3 display-block"
            type="button"
          >
            <span
              class="oec-text-with-icon text-underline"
            >
              Download Excel template
            </span>
          </button>
          <button
            class="usa-button usa-button--unstyled text-bold margin-bottom-3 display-block"
            type="button"
          >
            <span
              class="oec-text-with-icon text-underline"
            >
              Download CSV template
            </span>
          </button>
        </div>
        <div
          class="display-block"
        >
          <p>
            You can see a completed example template here:
          </p>
          <button
            class="usa-button usa-button--unstyled text-bold margin-bottom-3 display-block"
            type="button"
          >
            <span
              class="oec-text-with-icon text-underline"
            >
              Download Excel sample data
            </span>
          </button>
          <button
            class="usa-button usa-button--unstyled text-bold margin-bottom-3 display-block"
            type="button"
          >
            <span
              class="oec-text-with-icon text-underline"
            >
              Download CSV sample data
            </span>
          </button>
           
        </div>
      </div>
      <div
        class="grid-col-4"
        role="presentation"
      >
        <svg>
          PersonWithSpreadsheet.svg
        </svg>
      </div>
    </div>
    <div
      class="grid-row margin-top-2"
    >
      <a
        class="usa-button"
        href="/upload"
      >
        Go to file upload
      </a>
      <a
        class="usa-button"
        href="/roster"
      >
        Go to roster
      </a>
    </div>
  </div>
</DocumentFragment>
`;
